src/tchannel_bridge.js

```diff
@@ -0,0 +1,160 @@
+        // else start a span that is the child of the context span.
+        let childOf = context.openTracingSpan;
+        context.openTracingSpan = this._tracer.startSpan(endpoint, {
+            childOf: childOf
+        });
+        context.openTracingSpan.setTag(opentracing.Tags.PEER_SERVICE, req.serviceName);
+        context.openTracingSpan.setTag(opentracing.Tags.SPAN_KIND, opentracing.Tags.SPAN_KIND_RPC_CLIENT);
+        headers = this._injectSpan(context.openTracingSpan, headers);
+
+        // wrap callback so that span can be finished as soon as the response is received
+        let callback = this._tchannelCallbackWrapper.bind(null, context.openTracingSpan, wrappedCallback);
+
+        return wrappedSend.call(channel, req, endpoint, headers, body, callback);
+    }
+
+    _wrapTChannelRequest(channel, wrappedRequest, requestOptions) {
+        // We set the parent to a span with trace_id zero, so that tchannel's
+        // outgoing tracing frame also has a trace id of zero.
+        // This forces other tchannel implementations to rely on the headers for the trace context.
+        requestOptions.parent = { span: this._getTChannelParentSpan() };
+
+        let tchannelRequest = wrappedRequest.call(channel, requestOptions);
+        tchannelRequest.openTracingContext = requestOptions.openTracingContext;
+        return tchannelRequest;
+    }
+
+    /**
+     * A function that wraps a json, or thrift encoded channel, in order to populate
+     * the outgoing headers with trace context, and baggage information.
+     *
+     * @param {Object} channel - the encoded channel to be wrapped for tracing.
+     * @param {Object} context - A context that contains the outgoing span to be seralized.  If the context does not contain a span then a new root span is created.
+     * @returns {Object} channel - the trace wrapped channel.
+     * */
+    tracedChannel(channel: any): any {
+        let wrappedSend = channel.send;
+        let wrappedRequest = channel.channel.request;
+
+        // We are patching the top level channel request method, not the encoded request method.
+        channel.channel.request = this._wrapTChannelRequest.bind(this, channel.channel, wrappedRequest);
+
+        channel.send = this._wrapTChannelSend.bind(this, wrappedSend, channel);
+        return channel;
+    }
+
+    _getTChannelParentSpan(): TChannelSpan {
+        return {
+            id: [0, 0],
+            traceid: [0, 0],
+            parentid: [0, 0],
+            flags: 0
+        };
+    }
+
+    _extractSpan(operationName: string, headers: any): Span {
+        let traceContext: ?SpanContext = this._codec.extract(headers);
+        let options = {
+            childOf: traceContext
+        }
+        let span = this._tracer.startSpan(operationName, options);
+        return span;
+    }
+
+    _injectSpan(span: Span, carrier: any): any {
+        this._codec.inject(span.context(), carrier);
```

Comment:
there's little point in this being a method, it's just one statement, which is not reused anywhere